torch/_meta_registrations.py

```diff
@@ -51,6 +52,29 @@ def register(op):
     return wrapper
 
 
+def elementwise_meta(
+    *args,
+    type_promotion: ELEMENTWISE_TYPE_PROMOTION_KIND,
+):
+    # Perform type promotion, as this is expected from prim_metafunction
+    compute_dtype, result_dtype = utils.elementwise_dtypes(
+        *args,
+        type_promotion_kind=type_promotion,
+    )
+    args = [_maybe_convert_to_dtype(x, compute_dtype) for x in args]
+
+    # Broadcast
+    args = _maybe_broadcast(*args)
+
+    # Perform prim checks
+    type_promotion_prim = getattr(
+        ELEMENTWISE_PRIM_TYPE_PROMOTION_KIND, type_promotion.name
+    )
+    result = _prim_elementwise_meta(*args, type_promotion=type_promotion_prim)
+
+    return _maybe_convert_to_dtype(result, result_dtype)
```

Comment:
Does this ever do anything? Shouldn't `_prim_elementwise_meta` have done the type conversions already?

Reply:
It does. Consider that the inputs are `bfloat16`. Then the computation dtype is `float32`, but hte result_dtype is `bfloat16`.

Reply:
We shouldn't be using computation dtype for meta functions though

Reply:
I thought about that, but I was not sure whether the two are commutative.
I guess I could use `result_type` directly + `ELEMENTWISE_PRIM_TYPE_PROMOTION_KIND.DEFAULT` unconditionally.